pyccel/parser/semantic.py

```diff
@@ -492,11 +499,21 @@ def remove_variable(self, name):
     def get_header(self, name):
         """."""
         container = self.namespace
+        headers = []
         while container:
             if name in container.headers:
-                return container.headers[name]
+                headers += container.headers[name]
             container = container.parent_scope
-        return None
+        return headers
```

Comment:
Are you sure we need to look for the headers in multiple scopes? Can you give an example where this is necessary?

Reply:
this was the way we do it before, I can't think of a reason to look for them in multiple scopes. should I check only in the current scope

Reply:
I'm not sure. @saidctb ?

Reply:
@saidctb , this seems to have got lost. Do you know if we need to look for headers in multiple scopes? And if so, why?